tests/test_examples.py

```diff
@@ -1336,30 +1299,84 @@ def test_second_stage_optimization_CoilSet():
     np.testing.assert_allclose(field[0].current, 0, atol=1e-12)
 
 
-# TODO: replace this with the solution to Issue #1021
+@pytest.mark.slow
 @pytest.mark.unit
-def test_optimize_with_fourier_planar_coil():
-    """Test optimizing a FourierPlanarCoil."""
+def test_optimize_with_all_coil_types(DummyCoilSet, DummyMixedCoilSet):
+    """Test optimizing for every type of coil and dummy coil sets."""
+    sym_coils = load(load_from=str(DummyCoilSet["output_path_sym"]), file_format="hdf5")
+    asym_coils = load(
+        load_from=str(DummyCoilSet["output_path_asym"]), file_format="hdf5"
+    )
+    mixed_coils = load(
+        load_from=str(DummyMixedCoilSet["output_path"]), file_format="hdf5"
+    )
+    nested_coils = MixedCoilSet(sym_coils, asym_coils)
+    eq = Equilibrium()
+    # not attempting to accurately calc B for this test,
+    # so make the grids very coarse
+    quad_eval_grid = LinearGrid(M=2, sym=True)
+    quad_field_grid = LinearGrid(N=2)
+
+    def test(c, method):
+        target = 11
+        rtol = 1e-3
+        # first just check that quad flux works for a couple iterations
+        # as this is an expensive objective to compute
+        obj = ObjectiveFunction(
+            QuadraticFlux(
+                eq=eq,
+                field=c,
+                vacuum=True,
+                weight=1e-4,
+                eval_grid=quad_eval_grid,
+                field_grid=quad_field_grid,
+            )
+        )
+        optimizer = Optimizer(method)
+        (c,), _ = optimizer.optimize(c, obj, maxiter=2, ftol=0, xtol=1e-15)
+
+        # now check with optimizing geometry and actually check result
+        objs = [
+            CoilLength(c, target=target),
+        ]
+        extra_msg = ""
+        if isinstance(c, MixedCoilSet):
```

Comment:
Why do we only test curvature and torsion for `MixedCoilSet` and not other types of coils?

Reply:
Not sure. My thought was that it just wasn't needed. @dpanici any reason for this?

Reply:
MixedCoilSet contains the other types of coils, I did not want to reduce test coverage by removing the other test for it so I added it here.

Reply:
@dpanici but why not test them for all the coils? Removing this if statement should increase the test coverage, if anything

Reply:
They are expensive and this test already takes a while to run so I did not think it was necessary.

Reply:
As it is now, it takes 100s to run locally for me the test. If I change it around so that torsion and curvature are included with the quad flux for just that part of the test to ensure it runs without error, it takes way longer (5 minutes locally, checking again now to be sure it was not a fluke). That is really long, I'd rather not test it and trust that the logic works the same as the length. Or maybe just test ONE of them to cover the category of coil geometry objective that is not dim 1 like length is

Reply:
@ddudt torsion just adds too much time to the optimization, so I am fine with keeping the test as is.